### DISCUSSION TITLE
Open up midi for javascript?

### DISCUSSION THREAD

**Discussion by doozle7** [elgatosf/elgato#121] [Ideas]
Hi.

I'd like to do some fancy stuff with midi. I know there is an excellent midi plugin already but I've got some very specific ideas I'd like to implement. 

I could be wrong in my assumptions here, but I gather that the web server part of the StreamDeck is based on chromium. Midi in js is one of those 'secure context' permission things. I've poke around in javascript and I get the midi permissions for the StreamDeck are set to "deny". Could this be changed to "allow" or even "prompt" so that we can access midi via javascript?

That would be awesome as midi is an obvious use case for the StreamDeck.

Thanks in advance for your reply.

**Comment by ChekTek** [comment]
Hi @doozle7,

There are Midi plugins for Stream Deck available in [Elgato Marketplace](https://marketplace.elgato.com/search?q=midi), I don't know that this is something that we would consider adding directly to the SDK, but Stream Deck plugins are effectively small Node.js servers, which means to add Midi you could build a native addon using the [N-API](https://nodejs.org/api/n-api.html).
